beacon-chain/blockchain/process_block_test.go

```diff
@@ -1800,3 +1801,650 @@ func Test_verifyBlkFinalizedSlot_invalidBlock(t *testing.T) {
 	err = service.verifyBlkFinalizedSlot(wb)
 	require.Equal(t, true, IsInvalidBlock(err))
 }
+
+// See the description in #10777 and #10782 for the full setup
```

Comment:
The tests added here cover the main circumstances of a single branch being pruned and no viable head being available. It checks the behavior when the INVALID return comes from NotifyNewPayload and from FCU. Note however that in order to trigger justification change, at least one INVALID block needs to be imported to forkchoice, so this requires at least one INVALID block to be returned as SYNCING/ACCEPTED by notify_newPayload. 

These tests do not cover initial sync which can be added if requested, but the paths seem to be contained to the execution engine functions, so it should be similar from `onBlock` to `onBlockBatch`.  They also do not cover rebooting from this condition. This is a bit more dangerous since our starting head would be non-existent.